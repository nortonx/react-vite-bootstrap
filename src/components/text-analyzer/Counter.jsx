import PropTypes from "prop-types";
const Counter = ({ label, counter }) => {
  return(
    <div className="col" data-testid="counter">
      <h1 className="m-2 text-center">{label}</h1>
      <h2 className="text-center text-body-secondary">{counter}</h2>
    </div>
  );
};

Counter.propTypes = {
  label: PropTypes.string,
  counter: PropTypes.number
};

export default Counter;