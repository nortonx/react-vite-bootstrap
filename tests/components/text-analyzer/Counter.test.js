import { render, screen } from "@testing-library/react";

import Counter from "../../../src/components/text-analyzer/Counter";

describe("Counter", () => {
  beforeEach(() => {
    render(<Counter />);
  });

  it("Should render component and match snapshot", async () => {
    const counter = await screen.findByTestId("counter");
    expect(counter).toBeInTheDocument();
    expect(counter).toMatchSnapshot();
  })
})