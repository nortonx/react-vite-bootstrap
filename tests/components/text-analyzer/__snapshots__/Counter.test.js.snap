// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Should render component and match snapshot 1`] = `
<div
  class="col"
  data-testid="counter"
>
  <h1
    class="m-2 text-center"
  />
  <h2
    class="text-center text-body-secondary"
  />
</div>
`;
